src/shared/listeners/notify_users.py

```diff
@@ -30,32 +40,71 @@ def create_package_subscription_notifications(
         profile__package_subscriptions__overlap=affected_packages
     ).select_related("profile")
 
-    if not subscribed_users.exists():
-        logger.debug(f"No subscribed users found for packages: {affected_packages}")
-        return
+    # Find maintainers of affected packages from cached suggestion
+    maintainer_users = set()
+    if maintainers_github:
+        maintainer_users = set(
+            User.objects.filter(
+                username__in=maintainers_github,
```

Comment:
Would it be better to have another JOIN here instead of fetching all GitHub handles into Python just to send them back?